app/helpers/tickets_helper.rb

```diff
@@ -5,4 +5,16 @@ module TicketsHelper
   def sender_for_message(message)
     message.sender&.real_name || message.sender&.username || message.details[:sender_email]
   end
+
+  def successful_replies_in_reverse(_ticket, recipient)
+    successful_messages = @ticket.messages.reject do |message|
+      message.details[:delivery_failed]
```

Comment:
As we saw yesterday, a message can be both successful and have a failed delivery, so we should not reject a failed one if it also succeeded.

Reply:
That's true. I don't think we could do a `.select` for `:delivered` though as that would exclude the first message. What about:

```ruby
message.details[:delivered].nil? && message.details[:delivery_failed]
```

Reply:
Yeah, that sounds right.